lib/checks/aria/required-parent.js

```diff
@@ -1,5 +1,29 @@
+const { aria, dom } = axe.commons;
```

Comment:
I'd really like us to clean this up... my initial request for this was to create an `aria.getOwnerVirtual()`. I still think we should do this. Here's what this check would look like if we do this, which I think is far easier to read then what we have now:

```js
const role = aria.getRole(virtualNode)
const requiredContext = aria.requiredContext(role);
if (!requiredContext) {
  return true
}

// Accounts for role=presentation, as well as aria-owns
const owner = aria.getOwnerVirtual(virtualNode)
if (!owner) {
  return undefined;
}

const ownerRole = aria.getRole(owner)
if (requiredContext.some((contextRole) => contextRole === ownerRole)) {
  return true;
} else {
  this.relatedNode(owner);
  this.data(requiredContext);
  return false
}
```

Reply:
I would really like to keep this PR scope to fix the bug only & not go down the route of introducing nice to have's and helper function.

I will look into what the `getOwnerVirtual` entails.